File: report/src/main/java/com/henry/report/bean/Clicklog.java
package com.henry.report.bean;

/**
 * @Author: Henry
 * @Description: 点击流日志
 * @Date: Create in 2019/10/13 19:33
 **/

public class Clicklog {

    // 频道 ID
    private long channelID;
    // 产品的类别 ID
    private long categoryID ;
    // 产品 ID
    private long produceID ;
    // 用户 ID
    private long userID ;

    // 国家
    private String country;
    // 省份
    private String province;
    // 城市
    private String city;

    // 网络方式
    private String network;
    // 来源方式
    private String source;

    // 浏览器类型
    private String browserType;

    // 进入网站时间
    private Long entryTime ;
    // 离开网站实际
    private long leaveTime;

    public long getChannelID() {
        return channelID;
    }

    public void setChannelID(long channelID) {
        this.channelID = channelID;
    }

    public long getCategoryID() {
        return categoryID;
    }

    public void setCategoryID(long categoryID) {
        this.categoryID = categoryID;
    }

    public long getProduceID() {
        return produceID;
    }

    public void setProduceID(long produceID) {
        this.produceID = produceID;
    }

    public long getUserID() {
        return userID;
    }

    public void setUserID(long userID) {
        this.userID = userID;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getProvince() {
        return province;
    }

    public void setProvince(String province) {
        this.province = province;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getNetwork() {
        return network;
    }

    public void setNetwork(String network) {
        this.network = network;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getBrowserType() {
        return browserType;
    }

    public void setBrowserType(String browserType) {
        this.browserType = browserType;
    }

    public Long getEntryTime() {
        return entryTime;
    }

    public void setEntryTime(Long entryTime) {
        this.entryTime = entryTime;
    }

    public long getLeaveTime() {
        return leaveTime;
    }

    public void setLeaveTime(long leaveTime) {
        this.leaveTime = leaveTime;
    }

}

File: real-process/src/main/scala/com/henry/realprocess/bean/ClickLog.scala
package com.henry.realprocess.bean

import com.alibaba.fastjson.JSON

/**
  * @Author: Henry
  * @Description:
  * @Date: Create in 2019/10/20 14:45 
  **/

//频道ID（channelID）
//产品类别ID（categoryID）
//产品ID（produceID）
//国家（country）
//省份（province）
//城市（city）
//网络方式（network）
//来源方式（source）
//浏览器类型（browserType）
//进入网站时间（entryTime）
//离开网站时间（leaveTime）
//用户ID（userID）

case class ClickLog (
                   // 1、alt + 下拉
                   // 2、ctrl + shift + →，选中各个变量
                   var channelID:String,
                   var categoryID:String,
                   var produceID:String,
                   var country:String,
                   var province:String,
                   var city:String,
                   var network:String,
                   var source:String,
                   var browserType:String,
                   var entryTime:String,
                   var leaveTime:String,
                   var userID:String
                   )

object ClickLog{

  def apply(json: String): ClickLog = {

    // 先把json转换为JSONObject
    val jsonObject = JSON.parseObject(json)

    // 提取jsonObject中的各个属性，赋值给样例类
    var channelID = jsonObject.getString("channelID")
    var categoryID = jsonObject.getString("categoryID")
    var produceID = jsonObject.getString("produceID")
    var country = jsonObject.getString("country")
    var province = jsonObject.getString("province")
    var city = jsonObject.getString("city")
    var network = jsonObject.getString("network")
    var source = jsonObject.getString("source")
    var browserType = jsonObject.getString("browserType")
    var entryTime = jsonObject.getString("entryTime")
    var leaveTime = jsonObject.getString("leaveTime")
    var userID = jsonObject.getString("userID")

    ClickLog(
        channelID,
        categoryID,
        produceID,
        country,
        province,
        city,
        network,
        source,
        browserType,
        entryTime,
        leaveTime,
        userID
    )
  }
}